polus-precompute-slide-plugin/src/main.py

```diff
@@ -18,66 +102,96 @@ def main():
+    logger.info('images are stacked by variable {}'.format(stack_by))
+
     # Set up lists for tracking processes
     processes = []
     process_timer = []
     pnum = 0
-    
-    # Build one pyramid for each image in the input directory
-    # Each pyramid is built within its own process, with a maximum number of processes
-    # equal to number of cpus - 1.
-    for image in images:
-        if len(processes) >= multiprocessing.cpu_count()-1 and len(processes)>0:
-            free_process = -1
-            while free_process<0:
-                for process in range(len(processes)):
-                    if processes[process].poll() is not None:
-                        free_process = process
-                        break
-                time.sleep(3)
-                
-            pnum += 1
-            logger.info("Finished process {} of {} in {}s!".format(pnum,len(images),time.time() - process_timer[free_process]))
-            del processes[free_process]
-            del process_timer[free_process]
-        processes.append(subprocess.Popen("python3 build_pyramid.py --inpDir {} --outDir {} --image {}".format(input_dir,
-                                                                                                               output_dir,
-                                                                                                               image.name),
-                                                                                                               shell=True))
-        process_timer.append(time.time())
-    
-    # Wait for all processes to finish
-    while len(processes)>1:
-        free_process = -1
-        while free_process<0:
-            for process in range(len(processes)):
-                if processes[process].poll() is not None:
-                    free_process = process
-                    break
-            time.sleep(3)
-        pnum += 1
-        logger.info("Finished process {} of {} in {}s!".format(pnum,len(images),time.time() - process_timer[free_process]))
-        del processes[free_process]
-        del process_timer[free_process]
 
-    processes[0].wait()
-    
-    logger.info("Finished process {} of {} in {}s!".format(len(images),len(images),time.time() - process_timer[0]))
-    logger.info("Finished all processes!")
+    # variables to keep track of processes
+    im_count = 1
+    stack_count = 1
+
+    processstring = " --inpDir '{}' --outDir '{}' --pyramidType '{}' --imageType '{}'".format(input_dir, 
+                                                                                            output_dir,
+                                                                                            pyramid_type,
+                                                                                            image_type)
+
+    # if imagepattern or image type specified then images get stacked by the stack_by variables
+    if stack_by != None or imagepattern != None:
```

Comment:
The logic here seems overly complicated. It seems as though we should be able to just generate one process for each plane of a pyramid we want to write, regardless of whether that is saving things as separate pyramids or if they are different z-slices of one pyramid.